### DISCUSSION TITLE
Trying to create a UI Wrapper, but cant make it work with NextJS

### DISCUSSION THREAD

**Discussion by wallynm** [vercel/next.js#17798] [Help] [answered]
I have a repo, which is a wrapper-ui built around Chakra-ui
https://github.com/hypeguy-co/artisan

My main project, uses NextJS, and every time i try to import this repo it trows error saying i have multiple react projects.

What i can't understand i created two examples folder to test. One import with Webpack alone, and another with NextJS SSR, and turns out that only the NextJS version wont work...

I Think may related with some kind of the SSR part but i've already tried several options and none of they work.

The repo i've linked have both examples.
The`chakra-ui`  it's fully compatible with NextJS used alone, without this wrapper i just created.

It seems that i need to override alias been used by the Node/Webpack of NextJS but i've already implemented the alias rules inside Next, but none of they work as i expected.

**Comment by jamesmosier** [comment, marked as answer]
You are importing React multiple times. Once in your Next app and another in your [example-webpack](https://github.com/hypeguy-co/artisan/blob/f50fad3389bc840453863c4f80cc87963fcbd605/example-webpack/package.json#L34) project which is a dependency and therefore included with your library. 

You'd want to include `react` and `react-dom` as [`peerDependency`](https://classic.yarnpkg.com/en/docs/dependency-types/#toc-peerdependencies) in your `example-webpack` project.

You may also want to look at the [with-yarn-workspaces](https://github.com/vercel/next.js/tree/canary/examples/with-yarn-workspaces) example which has separate "packages" that are imported into a Next app.

**Reply by wallynm** [reply]
Hey @jamesmosier, thanks by the help!
Actually i've tried several ways to alias react and prevent it bundling together React module, none of they worked with Next. On the other hand, wiith pure webpack they work correctly, as pointed into the example repo. Seems an issue with the SSR part but didn't figured out the root cause.

Im saying it might have something related with SSR, because if i lazy load the component, it works!
Like the webpack example, so it might have some relation with this part, i think the moment i create the alias, NodeJS already parsed the module and points out there's two react versions...


Well, anyway i'll try to implement the yarn workspaces you've mentioned, never tried and might be fit with what i need.

**Reply by wallynm** [reply]
@jamesmosier look, for real i wasn't looking into a monorepo approach, but turns out that this solved me a lot of problems...
Thank you very much with this solution!!!

Earlier one of the solutions i've found was related with Lerna, but implementing it or nx was way over what i was looking for, turns out that workspaces was so easy to implement and indeed solved my issue! Thank you sooo much!!!
